homeworks/franktoth/4_exceptions_logging/to_do.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import os
+
+# Create the tasks.txt file if it doesn't exist
+if not os.path.exists("tasks.txt"):
+    with open("tasks.txt", "w") as f:
+        pass
+
+# Configure logging to write to both console and a log file
+logging.basicConfig(filename='todo.log', level=logging.INFO, format='%(asctime)s - %(message)s')
+
+#Reads tasks from a file.
+def read_tasks(filename):
+
+    try:
+        with open(filename, 'r') as f:
+            tasks = f.readlines()
+            return [task.strip() for task in tasks]
+    except FileNotFoundError:
+        logging.error("File not found.")
+        return []
+    except Exception as e:
+        logging.error(f"Error reading file: {e}")
+        return []
+
+#Adds a task to the file.
+def add_task(filename, task):
+
+    try:
+        with open(filename, 'a') as f:
+            f.write(task + '\n')
+        logging.info(f"Task added: {task}")
+    except Exception as e:
+        logging.error(f"Error writing to file: {e}")
+
+#Removes a task from the file by index.
+def remove_task(filename, task_index):
+
+    try:
+        tasks = read_tasks(filename)
+        del tasks[task_index]
+        with open(filename, 'w') as f:
+            for task in tasks:
+                f.write(task + '\n')
+        logging.info(f"Task removed: {task_index}")
+    except IndexError:
+        logging.error("No task with that index.")
+    except Exception as e:
+        logging.error(f"Error removing task: {e}")
+
+#Display the Task Manager menu
+def display_menu():
+    print("Task Manager")
+    print("1. Add Task")
+    print("2. View Tasks")
+    print("3. Remove Task")
+    print("4. Exit")
+
+filename = "tasks.txt"
```

Comment:
innen lefelé az egész bekerülhetne egy main() function-be, jelezvén hogy ez a kiinduló pontja a programodnak.
A filename változó lehetne inkább file_path, mert ez a file is a saját mappádba kéne kerüljön, nem a root-ba, így egy relative_path lesz az értéke.

Reply:
Megcsináltam mindkét modosítást.